### PR TITLE
Add bare bones eslint target and hook it up to travis

### PR DESCRIPTION
**Pull request by wlach**
[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/753)

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -220,6 +222,14 @@ module.exports = function(grunt) {
...
                 }
             }
+        },
+        eslint: {
+            options: {
+                config: '.eslintrc',
+                reset: true
+            },
+            target: ['ui/js/*.js',
+                     'ui/js/**/*.js']
```
Comment (edmorley) [inline Gruntfile.js]: Think we also need `ui/js/**/**/*.js` to include the deeper subdirectories under the `directives` directory.

**Comment by edmorley** [comment]
++ to adding a JS linter to Travis. We do already have jshint configs in-repo, but some rudimentary googling shows eslint to be preferred by some - so perhaps it's a better option? Should we perhaps remove the jshint configs? @camd does your PyCharm project use those jshint configs? Can it be made to use eslint instead?

**Comment by camd** [comment]
My PyCharm isn't currently using that file, no.  I think I was with an earlier version.  I'm cool to move to whatever.  :)

**Comment by wlach** [comment]
I'm inclined to remove the jshint commits as part of this, if that's ok. I think it'd be confusing to support more than one tool here.

File: docs/ui/installation.rst [hunk 2]
```diff
@@ -34,13 +34,27 @@ The sample configuration makes the UI load job/push data from the production ser
 
 If you wish to run the full treeherder Vagrant project (service + UI), remember to remove local.conf.js or else change ``thServiceDomain`` within it to refer to ``vagrant``, so the UI will use the local Vagrant service API.
 
+Validating JavaScript
+=====================
+
+We run our JavaScript code in the frontend through eslint_ to ensure
+that new code doesn't suffer from common errors. Before submitting a
+patch, check that your code passes these tests
```
Comment (tojonmz) [inline docs/ui/installation.rst]: Dunno perhaps a trailing period here, it's seems to be the paragraph style for this file.

Comment (wlach) [inline docs/ui/installation.rst]: Addressed in the final commit :)

**Comment by wlach** [comment]
I removed the jshint files in the final commit, to reduce confusion.

**Pull request merged**